agenthub/codeact_agent/codeact_agent.py

```diff
@@ -194,7 +194,7 @@ def step(self, state: State) -> Action:
             'temperature': 0.0,
         }
 
-        if self.llm.supports_prompt_caching:
+        if self.llm.enable_prompt_caching:
             params['extra_headers'] = {
```

Comment:
Anthropic announced, that prompt caching is out of beta, so this `extra_headers` could/should be removed.
At least that's what I did in one of my PR's, but haven't specifically tested caching yet.

Reply:
I didn't see any relevant announcements on the official docs, it's still in the beta phase. I confirmed this from 

https://docs.anthropic.com/en/docs/build-with-claude/prompt-caching . BTW, I found some interesting details in this 

doc, like system prompts must meet specific length requirements (exceeding 1024 tokens for Claude-3.5) to be eligible

 for caching, there are a few other quirks to keep in mind too. We need to get a bit more granular with how we handle 

prompt caching - it's not as straightforward as it might seem at first glance.

Reply:
From the anthropic doc (https://docs.anthropic.com/en/docs/build-with-claude/prompt-caching), the extra_headers `anthropic-beta: prompt-caching-2024-07-31` is still required. BTW, there are some interesting details in this doc, which indicates we may need a more refined prompt caching control.

Reply:
Glad you pointed that out! They took the 8192 output token size out of beta, not the prompt caching, I somehow misremembered that.
We might have to work on more details with caching, but can probably do that in a followup PR, too.

I hope I can test your branch out later today for review.

Reply:
yeah, prompt caching has a very significant token-saving meaning for the agent's long-context reasoning process, and I hope I can get involved in this work too.
Thanks for your review.